src/modules/Dashboard/reducers/dashboard.ts

```diff
@@ -6,16 +6,26 @@ const initialState = {
 	activeImageId: null,
 	isPending: true,
 	activeImagesStatus: 'all',
+	displaying: {
+		textFilter: '',
+	},
 };
 
 export default handleActions({
-	[types.SELECT_IMAGE]: (state: any , action: any) => ({
+	[types.SELECT_IMAGE]: (state: any, action: any) => ({
 		...state,
-		activeImageId: action.payload,
+		activeImageId: action.payload.id,
 		isPending: false,
 	}),
 	[types.SET_ACTIVE_IMAGES_STATUS]: (state: any , action: any) => ({
 		...state,
 		activeImagesStatus: action.payload,
 	}),
+	[types.UPDATE_TEXT_FILTER]: (state: any , action: any) => ({
+		...state,
+		displaying: {
+			...state.displaying,
+			textFilter: action.payload.text,
```

Comment:
textFilter: text, or change param name in action, and write:

displaying: {
			...state.displaying,
			textFilter,
}

this is much cleaner then you write